File: src/main/java/dev/vrba/simp/command/fun/animals/endpoints/BirdsApiEndpoint.java
package dev.vrba.simp.command.fun.animals.endpoints;

import dev.vrba.simp.command.fun.animals.AnimalApiEndpoint;
import org.jetbrains.annotations.NotNull;
import reactor.core.publisher.Mono;

import java.io.InputStream;
import java.util.Set;

public class BirdsApiEndpoint extends AnimalApiEndpoint {
    @Override
    @NotNull
    public Set<String> getNames() {
        return Set.of("bird", "birb", "birds", "birbs");
    }

    @Override
    @NotNull
    public String getUrl() {
        return "https://some-random-api.ml/img/birb";
    }

    @Override
    @NotNull
    public String getTitle() {
        return "Tweet!";
    }

    @Override
    @NotNull
    public Mono<String> extractImageFromResponse(@NotNull Mono<InputStream> response) {
        return this.extractImageFromJson(response, json -> json.get("link").asText());
    }
}

File: Services/ImageApi/Endpoints/BirdsApiEndpoint.cs
using System.Net.Http.Json;

namespace SimpBot.Services.ImageApi.Endpoints;

public class BirdsApiEndpoint : ImageApiEndpoint
{
    public override IEnumerable<string> Names { get; } = new[] {"bird", "birb", "birds", "birbs"};

    public override string Url => "https://some-random-api.ml/img/birb";

    public override async Task<string?> ExtractImageUrlAsync(HttpResponseMessage response)
    {
        var data = await response.Content.ReadFromJsonAsync<IDictionary<string, string>>();
        var image = data?["link"];

        return image;
    }
}